core/src/main/java/com/netflix/iceberg/avro/AvroFileAppender.java

```diff
@@ -56,6 +59,18 @@ public Metrics metrics() {
     return new Metrics(numRecords, null, null, null);
   }
 
+  @Override
+  public long length() {
+    if (stream != null) {
```

Comment:
The API make the contract that this is valid only when the file is closed. Would it make sense to check that in Avro and Orc file appenders?

Reply:
Yes.

Reply:
Fixed. Please have another look.